src/dtgtk/thumbnail.c

```diff
@@ -226,6 +226,32 @@ static void _thumb_draw_image(dt_thumbnail_t *thumb, cairo_t *cr)
 
   // and eventually the image border
   gtk_render_frame(context, cr, 0, 0, w * darktable.gui->ppd_thb, h * darktable.gui->ppd_thb);
+
+  // if needed we draw the working msg too
+  if(thumb->busy)
+  {
+    PangoRectangle ink;
+    PangoLayout *layout;
+    PangoFontDescription *desc = pango_font_description_copy_static(darktable.bauhaus->pango_font_desc);
+    const float fontsize = DT_PIXEL_APPLY_DPI(14);
+    pango_font_description_set_absolute_size(desc, fontsize * PANGO_SCALE);
+    pango_font_description_set_weight(desc, PANGO_WEIGHT_BOLD);
+    layout = pango_cairo_create_layout(cr);
+    pango_layout_set_font_description(layout, desc);
+    pango_layout_set_text(layout, _("working..."), -1);
+    pango_layout_get_pixel_extents(layout, &ink, NULL);
+    const float xc = w * darktable.gui->ppd_thb / 2.0,
+                yc = h * darktable.gui->ppd_thb * 0.85 - DT_PIXEL_APPLY_DPI(30), wd = ink.width * .5f;
+    cairo_move_to(cr, xc - wd, yc + 1. / 3. * fontsize - fontsize);
+    pango_cairo_layout_path(cr, layout);
+    cairo_set_line_width(cr, 2.0);
+    dt_gui_gtk_set_source_rgb(cr, DT_GUI_COLOR_LOG_BG);
+    cairo_stroke_preserve(cr);
+    dt_gui_gtk_set_source_rgb(cr, DT_GUI_COLOR_LOG_FG);
+    cairo_fill(cr);
+    pango_font_description_free(desc);
+    g_object_unref(layout);
```

Comment:
If this is a direct copy from another file, it would be better to refactor as a separate function called from both places.
Should that not be practical, there at least needs to be a comment on each copy referring to the other, so that future maintainers know that a fix needs to be applied in more than one place.

Reply:
Would be nice to refactor I agree.

Reply:
done